ChatAPI.Application/UseCases/Implementations/FriendshipService.cs

```diff
@@ -21,20 +21,20 @@ public class FriendshipService(IUserRepository userRepo, IFriendshipRepository f
 		/// </summary>
 		/// <param name="senderUserId">The user id which sends the friend request.</param>
 		/// <param name="phone">The receiver's phone number. </param>
-		public async Task<Result> AddFriend(int senderUserId, string phone)
+		public async Task<Result<FriendshipDTO>> AddFriend(int senderUserId, string phone)
 		{
 			User? targetUser = await userRepo.GetUserNoTracking(phone);
 
 			if (targetUser is null)
-				return Result.Failure("User with that phone is not found.", HttpStatusCode.NotFound);
+				return Result<FriendshipDTO>.Failure("User with that phone is not found.", HttpStatusCode.NotFound);
 
 			if (senderUserId == targetUser.Id)
-				return Result.Failure("You cannot add yourself as a friend.");
+				return Result<FriendshipDTO>.Failure("You cannot add yourself as a friend.");
```

Comment:
If this check is the first one, is there an option it will break something.

Reply:
Can you elaborate?